packages/workspace/src/tasks-runner/cache.ts

```diff
@@ -190,14 +190,19 @@ export class Cache {
     }
 
     return new Promise((res, rej) => {
-      execFile('cp', ['-a', src, dirname(directory)], (error) => {
-        if (!error) {
-          res();
-        } else {
-          this.useFsExtraToCopyAndRemove = true;
-          copy(src, directory).then(res, rej);
+      execFile(
```

Comment:
This will basically reintroduce code execution, as outputPath is not sanitized.

From https://nodejs.org/api/child_process.html#child_processexecfilesyncfile-args-options:

> If the `shell` option is enabled, do not pass unsanitized user input to this function. Any input containing shell metacharacters may be used to trigger arbitrary command execution.

Reply:
What alternative would you suggest? The current solution does not work on Windows.

We can perhaps try something like https://www.npmjs.com/package/cross-spawn that only targets Windows environment.

Reply:
I'm not sure I understand why this is failing on Windows, and I don't have that OS at hand debug. Is it expecting a full path to the file?

Reply:
Absent that, maybe at the least we can set `shell: true` only when running in Windows? That would at least limit the code execution to that operating system.

Reply:
It's expecting full file name e.g. `npx.cmd` if `shell` option is not set.

Reply:
Copy and remove can stay as they are, but web file server still needs a fix

Reply:
Alright, as long as it's clear that this is introducing the vulnerability back.